autosklearn/data/feature_validator.py

```diff
@@ -308,17 +308,6 @@ def _check_data(
                                      "limitation on scikit-learn being addressed via: "
                                      "https://github.com/scikit-learn/scikit-learn/issues/17123)"
                                      )
-            if self.enc_columns and self.enc_columns != enc_columns:
```

Comment:
Hey, is there a reason you removed this check?

Reply:
This is linked to the message above. One cannot reach this point in the code, that is, this check is unnecessary. Why is so? Well we did not have before the fit/transform interface. We only fit once and transform then onwards.

If you re-fit the whole data, then new data may be available and we should just change all what fit encompasses.  If we do not want to allow fit to be called multiple times, then let me know and I can add a check for this that will be more verbose about this.